### PR TITLE
CHET-587: Add start epoch to RDS secret 

### PR DESCRIPTION
**Pull request by dylanrathbone**
**PROPOSAL**
Add start epoch to RDS secret to avoid clashes on new migrations

**NOTE:** This is a non GA bug, but has the potential to be a real pain if encountered out in the wild.

🚧 Testing now ....

### CODE REVIEW

**Comment by dylanrathbone** [comment]
> Unfortunately there is a semantic coupling between the template and the secret name. Anything we put in the secret name needs to be deterministic or shareable with the migrationstack. We could do this by making it a parameter to the migration stack e.g. parameter: SecretName

:( Felt a bit easy...

**Comment by bpar476** [comment]
> Unfortunately there is a semantic coupling between the template and the secret name. Anything we put in the secret name needs to be deterministic or shareable with the migrationstack. We could do this by making it a parameter to the migration stack e.g. parameter: SecretName

In the template: 
`SECRET_PASSWORD=`aws secretsmanager  get-secret-value --secret-id ${SecretIdentifier} --region ${AWS::Region} --output text --query "SecretString"`
...
`SecretIdentifier: !Sub "atl-${AWS::StackName}-app-rds-password"`

**Comment by dylanrathbone** [comment]
> > Unfortunately there is a semantic coupling between the template and the secret name. Anything we put in the secret name needs to be deterministic or shareable with the migrationstack. We could do this by making it a parameter to the migration stack e.g. parameter: SecretName
> 
> In the template:
> 
> ```
> SECRET_PASSWORD=`aws secretsmanager  get-secret-value --secret-id ${SecretIdentifier} --region ${AWS::Region} --output text --query "SecretString"`
> ```
> 
> ...
> `SecretIdentifier: !Sub "atl-${AWS::StackName}-app-rds-password"`

Of course. Unfortunately we have no clear way of sharing the start epoch that is set against the secret name here.

**Comment by dylanrathbone** [comment]
> > > Unfortunately there is a semantic coupling between the template and the secret name. Anything we put in the secret name needs to be deterministic or shareable with the migrationstack. We could do this by making it a parameter to the migration stack e.g. parameter: SecretName
> > 
> > 
> > In the template:
> > ```
> > SECRET_PASSWORD=`aws secretsmanager  get-secret-value --secret-id ${SecretIdentifier} --region ${AWS::Region} --output text --query "SecretString"`
> > ```
> > 
> > 
> > ...
> > `SecretIdentifier: !Sub "atl-${AWS::StackName}-app-rds-password"`
> 
> Of course. Unfortunately we have no clear way of sharing the start epoch that is set against the secret name here.

Just noticed your comment on adding a parameter to the migration stack... Sounds like an option

**Comment by theghostwhoforks** [comment]
+1 to what @bpar476 said. The cleanest way to do this is to generate a random text in the key name and pass it in the parameter. 

This would be a very good fix to have. The secret name collision does not trigger valid state machine transitions and breaks a migration

File: core/src/main/java/com/atlassian/migration/datacenter/core/aws/db/restore/TargetDbCredentialsStorageService.java [hunk 1]
```diff
@@ -89,6 +89,6 @@ public void storeCredentials(String password) {
 
     public String getSecretName() {
         MigrationContext context = migrationService.getCurrentContext();
-        return String.format("atl-%s-migration-app-rds-password", context.getApplicationDeploymentId());
+        return String.format("atl-%s-migration-app-rds-password-%s", context.getApplicationDeploymentId(), context.getStartEpoch());
```
Comment (theghostwhoforks) [inline core/src/main/java/com/atlassian/migration/datacenter/core/aws/db/restore/TargetDbCredentialsStorageService.java:92]: would the resulting length be in compliance with the key name restrictions put in by AWS Param store?

**Comment by bpar476** [comment]
Can we close this @dylanrathbone ?

**Comment by bpar476** [comment]
There's been no activity here for a while. I'm going to close this PR and we can look back at it if we decide to address the issue in planning.

**Pull request closed without merging**